### ISSUE TITLE
Another memory corruption problem?

### ISSUE THREAD

**Issue by fgsch** [tinygo-org/tinygo#1990]
The code below prints:

`len(b)=55265: "foo" -> ""`

Tinygo version:

`tinygo version 0.19.0 darwin/amd64 (using go version go1.16.5 and LLVM version 11.0.0)`

Code:

```
package main

import (
	"fmt"
	"os"
)

func main() {
	pre := os.Getenv("PATH")
	b := make([]byte, 55265)
	post := os.Getenv("PATH")
	fmt.Printf("len(b)=%d: %q -> %q\n", len(b), pre, post)
}
```

Compilation and testing:

- `tinygo build -target=wasi -o bug.wasm bug.go`
- `wasmtime run  bug.wasm --env PATH=foo`

Changing the len of b will affect the output.

~This might be a duplicate of #1966.~

**Comment by dkegel-fastly** [comment]
Thanks for finding a minimal reproducer!

**Comment by aykevl** [comment]
Indeed, this minimal reproducer made it much easier to spot the underlying issue!

I think I know the bug, this should be a fix: https://github.com/tinygo-org/tinygo/pull/1992
I'm not sure whether this is related to #1966 but if you can test it, that would be great.

**Comment by fgsch** [comment]
Thanks. #1992 fixes the issue in this PR but #1966 still fails.

**Comment by fgsch** [comment]
Credits should go to @peterbourgon for repro.

**Comment by fgsch** [comment]
This can be marked as next-release now too 🎉

**Comment by deadprogram** [comment]
Released with 0.20.0 so now closing. Thank you!
